### PR TITLE
Fix race condition issue

### PR DESCRIPTION
**Pull request by TommyLike**
For #359

### CODE REVIEW

File: pkg/controllers/job/job_controller_actions.go [hunk 1]
```diff
@@ -253,6 +253,7 @@ func (cc *Controller) syncJob(jobInfo *apis.JobInfo, updateStatus state.UpdateSt
 
 	waitCreationGroup := sync.WaitGroup{}
 	waitCreationGroup.Add(len(podToCreate))
+	stateMutex := sync.Mutex{}
```
Comment (k82cn) [inline pkg/controllers/job/job_controller_actions.go]: move mutex into function

Comment (TommyLike) [reply pkg/controllers/job/job_controller_actions.go]: `classifyAndAddUpPodBaseOnPhase` is not the only place that we need to protect.

Comment (k82cn) [reply pkg/controllers/job/job_controller_actions.go]: the other also should move to function, do not put mutex in more than on place.

Comment (hzxuzhonghu) [reply pkg/controllers/job/job_controller_actions.go]: FYI：
```
 // A Mutex must not be copied after first use.

```


So this is not right, you can change to stateMutex := &sync.Mutex{}

Comment (TommyLike) [reply pkg/controllers/job/job_controller_actions.go]: @hzxuzhonghu  just curious, for which reason why this is not working? Which code leads to the multiple copies of mutex instance?

Comment (hzxuzhonghu) [reply pkg/controllers/job/job_controller_actions.go]: Actually, this var is used by many go routines, and if you print the address of it in the go routines, you can find they are not same.  So can not protect the critical-section using different lock.

Comment (TommyLike) [reply pkg/controllers/job/job_controller_actions.go]: But the mutex instance is not used as a parameter for those go routines.

Comment (hzxuzhonghu) [reply pkg/controllers/job/job_controller_actions.go]: make sense, my memory messed up.

**Comment by TommyLike** [comment]
@asifdxtreme  what's the command to retest the patch?

/retest

**Pull request merged**